src/crewai/agent.py

```diff
@@ -114,39 +114,42 @@ class Agent(BaseAgent):
     )
 
     @model_validator(mode="after")
-    def set_agent_ops_agent_name(self) -> "Agent":
-        """Set agent ops agent name."""
+    def post_init_setup(self):
+        # Set agent_ops_agent_name
         self.agent_ops_agent_name = self.role
-        return self
 
-    @model_validator(mode="after")
-    def set_agent_executor(self) -> "Agent":
-        """Ensure agent executor and token process are set."""
+        # Set up LLM callbacks
         if hasattr(self.llm, "model_name"):
-            token_handler = TokenCalcHandler(self.llm.model_name, self._token_process)
+            self._setup_llm_callbacks()
 
-            # Ensure self.llm.callbacks is a list
-            if not isinstance(self.llm.callbacks, list):
-                self.llm.callbacks = []
+        # Set up agent executor
```

Comment:
Just a small note: The code is pretty self-explanatory. In such cases, I try to avoid commenting. What do you think?

Reply:
Agreed! Will remove!